.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,37 @@
+name: CI Test Suite
+
+on: 
+    push:
+        branches: 
+          - main
+    pull_request: 
+        branches:
+          - main
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+
+            - name: Setup go
+              uses: actions/setup-go@v5
+              with:
+                  go-version: 1.23
+
+            - name: Install Dependencies
+              run: go mod tidy
+            
+            - name: Run Tests
+              run: go test ./... -json | tee test-results.log
+            
+            - name: Install Test Report
+              run: go install github.com/becheran/go-testreport@v0.2.2
+            
+            - name: Generate Test Report
```

Comment:
Follow [the appropriate instructions](https://github.com/becheran/go-testreport?tab=readme-ov-file#github-actions) to setup the GitHub Step Summary.